camel/Camel.php

```diff
@@ -126,24 +126,18 @@ public static function getEndpointIndexInGroup(array $groups, OutputEndpointData
 
     /**
      * @param array[] $endpoints
-     * @param array $endpointGroupIndexes Mapping of endpoint IDs to their index within their group
      *
      * @return array[]
      */
-    public static function groupEndpoints(array $endpoints, array $endpointGroupIndexes): array
+    public static function groupEndpoints(array $endpoints): array
     {
         $groupedEndpoints = collect($endpoints)
             ->groupBy('metadata.groupName')
             ->sortKeys(SORT_NATURAL);
 
-        return $groupedEndpoints->map(function (Collection $endpointsInGroup) use ($endpointGroupIndexes) {
+        return $groupedEndpoints->map(function (Collection $endpointsInGroup) {
             /** @var Collection<(int|string),ExtractedEndpointData> $endpointsInGroup */
-            $sortedEndpoints = $endpointsInGroup;
-            if (!empty($endpointGroupIndexes)) {
-                $sortedEndpoints = $endpointsInGroup->sortBy(
```

Comment:
You shouldn't remove this condition. Sorting by `$endpointGroupIndexes` allows us to retain the user's sort order, if they have made changes to the Camel files. See https://scribe.knuckles.wtf/laravel/documenting/custom-endpoints#scenario-2-sorting-your-endpoints-or-groups

A better option would be to sort by `$endpointGroupIndexes` if present, or else sort by the URL.

Reply:
Also, I'm not even sure that we need this change. I don't think #389 is an issue at all. `$endpointGroupIndexes` is only present **if** you have run Scribe before. If you haven't, Scribe will not sort endpoints, which means you will get Laravel's default sort. Sorry, but I think this (and that issue) should be closed (unless someone can prove I'm wrong🤔).

The only issue I can see is that, if you add a new endpoint *after* running Scribe, it will automatically be put at the end of that group. But you can reset the sort order to the default with `--force`.

Reply:
Yeah, I missed this part about Camel. But, unfortunately, sorting at the current moment  does not work "in alphabetical order".
Routes are generated from Router::getRoutes, which are not sorted. Therefore, normal (alphabetical) sorting does not happen.
I think we need to sort these paths in
https://github.com/knuckleswtf/scribe/blob/456d9204ef326366d0285d30d8db66197ab253fa/src/Matching/RouteMatcher.php#L21
Not after Camel processing.

Reply:
Alright, I see your point. But the PR is a bit to noisy for me to review. Please revert the endpointId changes, and add only the sorting.

Then, please add a small test that verifies that this sort order is kept, something like this:

```php
    /** @test */
    public function sorts_endpoints_by_uri()
    {
        RouteFacade::get('/api/industries', [TestController::class, 'dummy']);
        RouteFacade::get('/api/languages', [TestController::class, 'dummy']);
        RouteFacade::post('/api/languages', [TestController::class, 'dummy']);
        RouteFacade::get('/api/industries/{industry}', [TestController::class, 'dummy']);
        RouteFacade::get('/api/languages', [TestController::class, 'dummy']);
        RouteFacade::post('/api/industries/{industry}', [TestController::class, 'dummy']);
        RouteFacade::get('/api/languages/{language}', [TestController::class, 'dummy']);

        $this->artisan('scribe:generate');

        // Verify the order 
        $crawler = new Crawler(file_get_contents('public/docs/index.html'));
        $expectedEndpoints = $crawler->filter('h2');
        $this->assertEquals("GET./api/industries", $expectedEndpoints->getNode(0)->textContent);
        $this->assertEquals("GET./api/industries/{industry}", $expectedEndpoints->getNode(1)->textContent);
        $this->assertEquals("POST./api/industries/{industry}", $expectedEndpoints->getNode(2)->textContent);
        // ..and so on
    }
```